packages/next/bin/next.ts

```diff
@@ -94,7 +94,12 @@ if (process.env.NODE_ENV && !standardEnv.includes(process.env.NODE_ENV)) {
 ;(process.env as any).NODE_ENV = process.env.NODE_ENV || defaultEnv
 
 // Make sure commands gracefully respect termination signals (e.g. from Docker)
-process.on('SIGTERM', () => process.exit(0))
+process.on('SIGTERM', () =>
```

Comment:
What do you think about these listeners being omitted when an environment variable is present and instead adding custom handling for this in `_document`? 

This would allow more control over calling `process.exit()` instead of relying on an arbitrary time value. e.g.

```js
if (!process.env.NEXT_MANUAL_SIGTERM) {
  process.on('SIGTERM', () => process.exit(0))
}
```

```js
// pages/_document.js

if (process.env.NEXT_MANUAL_SIGTERM) {
  // custom handling here
  process.on('SIGTERM', () => {
    console.log('cleaning up')
    process.exit(0)
  })
}
```

Reply:
Handling correctly SIGTERM to gracefully shutdown the next server should be a day 1 mandatory feature.

It's amazing that this is still not supported and that the only option is to go with a customized server, loosing some benefit of next (optimization)

Reply:
love this, we would like to overwrite this to close database connection ...